File: 21_01_14/leetcode-1018.cpp
#include<cstdio>
class Solution {
public:
    vector<bool> prefixesDivBy5(vector<int>& A) {
        int num = 0;
        int length = A.size();
        vector<bool> result;
        for(int i=0; i<length; i++){
            num = ((num << 1) + A[i]) % 5;
            result.emplace_back(num  ==0);
        }
        return result;
    }
};

File: 21_01_14/leetcode-1018.py
from typing import List
class Solution:
    def prefixesDivBy5(self, A: List[int]) -> List[bool]:
        result = []
        num = 0
        for i in A:
            num = num * 2 + i
            if num % 5 == 0:
                result.append(True)
            else:
                result.append(False)
        return result


def test_leetcode_1018():
    solution = Solution()
    A = [0, 1, 1]
    answer = [True, False, False]
    assert solution.prefixesDivBy5(A) == answer
